### PR TITLE
:sparkles: add repository functions

### LINKED ISSUE #120: Add support for repository management
**Is your feature request related to a problem? Please describe.**
We us the repository functionnality to give customized images to the users of our platforme.

We would like to automate repository update, but there is no documented API on that and it is not available with this library.

**Describe the solution you'd like**

It would be nice if we could:

- Create a repository
- Update a repository

With this library.

**Describe alternatives you've considered**
X

**Additional context**
X

### PR DESCRIPTION
**Pull request by Qiwei08**
Add repository functions 
close #120

### CODE REVIEW

File: saagieapi/repositories/repositories.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        NB: You can only list repository if you have at least the viewer role on the
+        catalog
+
+        Returns
+        -------
+        dict
+            Dict of repository ids, names and last reversible IDs
+        """
+
+        return self.saagie_api.client_gateway.execute(query=gql(GQL_LIST_REPOSITORIES_MINIMAL))
```
Comment (guillaumeprevost76) [inline saagieapi/repositories/repositories.py:45]: I think you should use fragment in your graphql request to merge these 2 functions.
You can show how it is done for the apps

File: saagieapi/saagie_api.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import re
 from typing import Dict, List, Tuple
 
+import deprecation
```
Comment (tquiviger) [inline saagieapi/saagie_api.py:5]: is it used?

File: saagieapi/saagie_api.py [hunk 3]
```diff
@@ -206,7 +209,7 @@ def get_cluster_capacity(self) -> Dict:
     # ###                    repositories                   ####
     # ##########################################################
 
-    def get_repositories_info(self) -> Dict:
+    def get_repositories_info(self):
```
Comment (TFellet) [inline saagieapi/saagie_api.py:212]: Why is Dict removed ?

File: saagieapi/repositories/gql_queries.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        }
+        ... on FailedSynchronization {
+          failure
+        }
+      }
+      lastReversibleId
+    }
+  }
+}
+"""
```
[the remark this answers is not in the document]
Comment (Qiwei08) [reply saagieapi/repositories/gql_queries.py]: good idea, done

File: saagieapi/repositories/repositories.py (hunk 1 above)
Comment (Qiwei08) [reply saagieapi/repositories/repositories.py]: done

File: saagieapi/saagie_api.py (hunk 2 above)
Comment (Qiwei08) [reply saagieapi/saagie_api.py]: the import is removed

File: saagieapi/repositories/repositories.py [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+        catalog
+
+        Returns
+        -------
+        dict
+            Dict of repository ids, names and last reversible IDs
+        """
+        params = {"minimal": True, "lastSynchronization": True}
+
+        return self.saagie_api.client_gateway.execute(query=gql(GQL_LIST_REPOSITORIES), variable_values=params)
```
Comment (guillaumeprevost76) [inline saagieapi/repositories/repositories.py:56]: This function is not necessary any more as we can proceed with the function list()

**Comment by TFellet** [comment]
There is no unit tests for repositories functions

File: saagieapi/repositories/repositories.py (hunk 5 above)
Comment (Qiwei08) [reply saagieapi/repositories/repositories.py]: Thanks 👍 , it is removed

**Review by guillaumeprevost76 — APPROVED** [review]

**Pull request merged**